ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -186,6 +187,20 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
     return new StoreInfo(messageReadSet, messageInfos);
   }
 
+  /**
+   * Returns the store info for the given ids asynchronously
+   * @param ids The list of ids whose messages need to be retrieved
+   * @param storeGetOptions A set of additional options that the store needs to use while getting the message
+   * @return a {@link CompletableFuture} that will eventually contain the {@link StoreInfo} for the given ids or the
+   *         {@link StoreException} if an error occurred
+   */
+  public CompletableFuture<StoreInfo> getAsync(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> storeGetOptions) {
```

Comment:
Can we have separate sync and asyn implementations for CloudBlobStore?

Reply:
Since we are only adding async implementations for methods that handle requests (such as get, put, etc) and methods related to replication (findEntriesSince, etc) are still synchronous, I was just wondering if adding async methods in the existing CloudBlobStore implementation would be sufficient. What do you think?